q2_demux/_summarize/_visualizer.py

```diff
@@ -53,46 +54,56 @@ def _link_sample_n_to_file(files, counts, subsample_ns):
     return results
 
 
-def _subsample_paired(fastq_map):
+def _subsample_paired(fastq_map, delta=1):
     qual_sample = collections.defaultdict(list)
-    seq_len = None
+    min_seq_len = math.inf
+    max_seq_len = 0
+    warning = ''
     for fwd, rev, index in fastq_map:
         file_pair = zip(_read_fastq_seqs(fwd), _read_fastq_seqs(rev))
         for i, (fseq, rseq) in enumerate(file_pair):
-            if seq_len is None:
-                seq_len = len(fseq[1])
-            if seq_len != len(fseq[1]):
-                raise ValueError(inconsistent_length_template
-                                 % (seq_len, len(fseq[1])))
-            if seq_len != len(rseq[1]):
-                raise ValueError(inconsistent_length_template
-                                 % (seq_len, len(rseq[1])))
+            min_seqs = min(len(fseq[1]), len(rseq[1]))
+            max_seqs = max(len(fseq[1]), len(rseq[1]))
+            min_seq_len = min(min_seq_len, min_seqs)
+            max_seq_len = max(max_seq_len, max_seqs)
             if i == index[0]:
                 qual_sample['forward'].append(_decode_qual_to_phred33(fseq[3]))
                 qual_sample['reverse'].append(_decode_qual_to_phred33(rseq[3]))
                 index.pop(0)
                 if len(index) == 0:
                     break
+    if abs(max_seq_len - min_seq_len) > delta:
+        warning = (inconsistent_length_template %
+                   (min_seq_len, max_seq_len))
+        for key in qual_sample.keys():
+            array_list = qual_sample[key]
+            for i in range(len(array_list)):
+                array_list[i] = array_list[i][0:min_seq_len:1]
+    return [qual_sample, warning]
 
-    return qual_sample
 
-
-def _subsample_single(fastq_map):
+def _subsample_single(fastq_map, delta=1):
     qual_sample = collections.defaultdict(list)
-    seq_len = None
+    min_seq_len = math.inf
+    max_seq_len = 0
+    warning = ''
     for file, index in fastq_map:
         for i, seq in enumerate(_read_fastq_seqs(file)):
-            if seq_len is None:
-                seq_len = len(seq[1])
-            if seq_len != len(seq[1]):
-                raise ValueError(inconsistent_length_template
-                                 % (seq_len, len(seq[1])))
+            min_seq_len = min(min_seq_len, len(seq[1]))
+            max_seq_len = max(max_seq_len, len(seq[1]))
             if i == index[0]:
                 qual_sample['forward'].append(_decode_qual_to_phred33(seq[3]))
                 index.pop(0)
                 if len(index) == 0:
                     break
-    return qual_sample
+    if abs(max_seq_len - min_seq_len) > delta:
+        warning = (inconsistent_length_template %
+                   (min_seq_len, max_seq_len))
+        for key in qual_sample.keys():
+            array_list = qual_sample[key]
+            for i in range(len(array_list)):
+                array_list[i] = array_list[i][0:min_seq_len:1]
```

Comment:
No worries!  It doesn't matter anyway post-the-conversation-we-just-had because we are now coloring the vis rather than truncating so I'm just removing that code and working on `summarize`.